packages/cdk/lib/stacks/engines/toil-engine-construct.ts

```diff
@@ -0,0 +1,104 @@
+import { Aws } from "aws-cdk-lib";
+import { IVpc } from "aws-cdk-lib/aws-ec2";
+import { FargateTaskDefinition, LogDriver } from "aws-cdk-lib/aws-ecs";
+import { ApiProxy, SecureService } from "../../constructs";
+import { IRole } from "aws-cdk-lib/aws-iam";
+import { createEcrImage, renderServiceWithTaskDefinition } from "../../util";
+import { Bucket } from "aws-cdk-lib/aws-s3";
+import { EngineOptions, ServiceContainer } from "../../types";
+import { LogGroup, ILogGroup } from "aws-cdk-lib/aws-logs";
+import { EngineOutputs, EngineConstruct } from "./engine-construct";
+import { ToilJobRole } from "../../roles/toil-job-role";
+import { ToilEngineRole } from "../../roles/toil-engine-role";
+import { IJobQueue } from "@aws-cdk/aws-batch-alpha";
+import { Construct } from "constructs";
+
+export interface ToilEngineConstructProps extends EngineOptions {
+  /**
+   * AWS Batch JobQueue to use for running workflows.
+   */
+  readonly jobQueue: IJobQueue;
+}
+
+export class ToilEngineConstruct extends EngineConstruct {
+  public readonly engine: SecureService;
+  public readonly adapterRole: IRole;
```

Comment:
If Toil can speak WES without needing to use a Lambda why is there an `adapterRole` and `adapterLogGroup`? Perhaps these need renaming?

Reply:
I left `adapterRole` and `adapterLogGroup` in under the impression that the type system required them to be there. Toil doesn't use them, and on looking at this again it seems like  `adapterRole` (now?) never makes it out to the base engine interface.

But [`EngineOutputs` always has an `adapterLogGroup` right now](https://github.com/aws/amazon-genomics-cli/blob/75603a860ca401b7be2d5d04c51248a227283d00/packages/cdk/lib/stacks/engines/engine-construct.ts#L12-L17), and the base `EngineConstruct` defines a method that has to return an `EngineOutputs`. I think to get rid of it I would need to not make `ToilEngineConstruct` an `EngineConstruct`, or else remove the field from the base `EngineOutputs` and duplicate it and any code that uses it in all the other engines. Or else add a new intermediate `AdaptedEngineOutputs` and `AdaptedEngineConstruct` and change the other engines to use those.